### DISCUSSION TITLE
Media queries and alternate versions

### DISCUSSION THREAD

**Discussion by giacomo-petri** [w3c/wcag#3784] [Q&A]
By detecting user preferences, developers and authors can create various iterations of a website.

As I've previously pointed out in a clarification of SC 2.2.2, by detecting user preferences, developers and authors have the ability to generate different "versions" of the website based on user preferences.

Let me outline two distinct scenarios:

Scenario 1:
Utilizing the "prefers-reduced-motion: reduce" media query, authors can opt to conceal significant informative animated segments of the page, thereby meeting SC 2.2.2. However, this action may restrict access to equivalent information for certain users, although no specific criterion fails at this juncture.

Scenario 2:
Consider a default "light version" of the website compliant with all WCAG success criteria, including SC 1.4.3 Contrast (Minimum). Introducing a "prefers-color-scheme: dark" media query allows authors to offer a dark mode version of the website, albeit failing SC 1.4.3.

These examples evoke contrasting sentiments for me.

In the first scenario, regarding the "prefers-reduced-motion: reduce" media query as an alternative version enables evaluators to satisfy SC 2.2.2 while still falling short of full WCAG compliance. This is because, if viewed as an alternate version, it fails to provide equal information to all users, thereby not meeting the definition of a "conforming alternate version."

Conversely, in the second scenario, employing the same approach, there exists a website version that fulfills all WCAG SC, despite the dark mode version not meeting SC 1.4.3. Considering the normative aspect of SC 1.4.3 and acknowledging its purpose to ensure minimum perceptibility without reliance on assistive technology, it could be argued that the dark mode should also meet this requirement, as users may require dark mode for device operation.

I find myself wrestling with conflicting emotions regarding this issue, especially considering the intricate challenge of testing all conceivable user preferences and combinations. Certain versions may adhere to WCAG requirements solely with specific preferences, while others might pass despite a specific version failing due to a particular preference that is widely shared among users (such as dark mode).
In my opinion, there are instances where it's understandable, like meeting 2.2.2 through reduced motion. However, it becomes unjustifiable if dark mode doesn't meet minimum contrast ratios.

These examples are deliberately exaggerated because I believe that a consistent approach is essential and should be addressed somewhere in WCAG to establish coherence among evaluators.

What are your thoughts on this?

**Comment by JAWS-test** [comment]
Scenario 2: related: https://github.com/w3c/wcag/issues/2889 and https://github.com/w3c/wcag/issues/2909

**Comment by mbgower** [comment]
@giacomo-petri you've opened a more specific issue that touches on scenario 1. 
This issue seems potentially appropriate as a discussion. How would you feel about convert it to a discussion?

**Comment by nattarnoff** [comment]
Carrying this over from the issue filed - as someone who helped drive initial discussions around prefers-reduced-motion and 2.3.3 I am happy to go on record that the media query alone is not a sufficient technique for 2.2.2.
First, the media query is for reducing motion, not to specifically eliminate it. Part of me wishes it did, but then a lot of content would be lost to those of us who rely on it. Which brings me to my second point, there should not be any grey in whether a technique is successful or a failure. Prefers-reduced-motion is there specifically FOR the grey areas. We know motion & animation can improve usability, but too much makes some people ill. The media query lets us have the best of both.
